pkg/sql/internal.go

```diff
@@ -896,6 +901,10 @@ func (ie *InternalExecutor) execInternal(
 	timeReceived := timeutil.Now()
 	parseStart := timeReceived
 	parsed, err := parser.ParseOne(stmt)
+	if tree.CanModifySchema(parsed.AST) && txn != nil && !ie.checkIfTxnBound() {
```

Comment:
nit: encapsulate the condition and error into the helper? `if err := ie.checkIfStmtIsAllowed(parsed, txn); err != nil { return nil, err }`?

Reply:
Done.